### PR TITLE
neutron: add rhv support, use intf regexp only if using MAC

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

**Comment by safchain** [comment]
run devstack-tests

**Comment by lebauce** [comment]
run skydive-ppc64le-tests

File: topology/probes/neutron/neutron.go [hunk 1]
```diff
@@ -126,10 +129,8 @@ func (mapper *NeutronProbe) retrievePort(portMd portMetadata) (port ports.Port,
 		}
 
 		for _, p := range portList {
-			if p.MACAddress == portMd.mac {
-				port = p
-				return false, nil
-			}
+			port = p
```
Comment (nplanel) [inline topology/probes/neutron/neutron.go]: if len(portList) >0 {
 port = portList[0]
 return false,nil
}

File: topology/probes/neutron/neutron.go [hunk 2]
```diff
@@ -203,27 +204,26 @@ func (mapper *NeutronProbe) nodeUpdater() {
 			continue
 		}
 
-		if mac, _ := node.GetFieldString("MAC"); mac == "" {
-			mapper.graph.RUnlock()
-			continue
+		name, _ := node.GetFieldString("Name")
+		if name == "" {
+			return
```
Comment (nplanel) [inline topology/probes/neutron/neutron.go]: mapper.graph.RUnlock()
return

File: topology/probes/neutron/neutron.go [hunk 3]
```diff
@@ -74,41 +74,41 @@ type attributes struct {
...
+	name   string
 	mac    string
 	portID string
 }
 
-// NeutronPortNotFound error
-type NeutronPortNotFound struct {
-	MAC string
-}
+var emptyPortMetadata = portMetadata{}
```
Comment (nplanel) [inline topology/probes/neutron/neutron.go]: const

File: topology/probes/neutron/neutron.go (hunk 2 above)
Comment (safchain) [reply topology/probes/neutron/neutron.go]: thx done

File: topology/probes/neutron/neutron.go (hunk 1 above)
Comment (safchain) [reply topology/probes/neutron/neutron.go]: done

File: topology/probes/neutron/neutron.go (hunk 3 above)
Comment (safchain) [reply topology/probes/neutron/neutron.go]: can't be a const

**Comment by safchain** [comment]
run skydive-functional-tests-backend-orientdb

File: topology/probes/neutron/neutron.go (hunk 3 above)
Comment (nplanel) [reply topology/probes/neutron/neutron.go]: yeah my bad

**Pull request merged**